### PR TITLE
Prepare 0.4.0 release

### LINKED ISSUE #52: Version release
@thejpster there hasn't been a release on crates.io for quite a long time. Some features on the HEAD, such as deletion of files is not accessible for those using the crate directly from the crates.io.

Is it possible to schedule a new release soon?

Kind regards

### PR DESCRIPTION
**Pull request by eldruin**
Summary:
- Bump version to 0.4.0
- Add custom error type for `File` operations currently with only one variant: `File::InvalidOffset`
- Split filesystem and fat module into submodules (no API changes)
- Extract `Controller` into submodule (no API changes)
- Small code improvements suggested by clippy
- Update env_logger dependency
- Update repository URL
- Remove Travis CI

Closes #52

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,17 +6,20 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-- Optionally use [defmt](https://github.com/knurling-rs/defmt) for logging.
-    Controlled by `defmt-log` feature flag.
+
+[Unreleased]: https://github.com/rust-embedded-community/embedded-sdmmc-rs/compare/v0.3.0...develop
```
Comment (thejpster) [inline CHANGELOG.md:10]: v0.4.0

**Review by thejpster — DISMISSED** [review]
One typo, otherwise excellent. Thank you!

File: CHANGELOG.md (hunk 1 above)
Comment (eldruin) [reply CHANGELOG.md]: Nice catch! it is fixed now.

**Review by thejpster — APPROVED** [review]
Assuming you only changed the one line, this is fine.

**Pull request merged**